Add ntf contract address to batch data

phipsae opened the pull request:
Hi Carlos,

This PR adds the NFT contract address to the batch data point.

I went for your suggested automatic approach, fetching the address using ethers on the backend. 
However, I was unsure whether to set the attribute to the zero address or an empty string when the address is unavailable. I decided to go with an empty string.

(There is no migration. We can just edit the batches and submit it to the database, then the data should be fetched.)

Let me know what you think.

Best,
Philip

carletex commented on packages/backend/routes/batches.js:
```diff
@@ -138,6 +140,10 @@ router.patch("/update", withRole("admin"), async (req, res) => {
 
   if (batchContractAddress) {
     batchData.contractAddress = batchContractAddress;
+    batchData.nftContractAddress = await getNFTContractAddress(batchContractAddress);
```
A bit of a nitpick:

This is being called every time you edit a batch, right? It doesn't matter if the batchContractAddress is the same. Since we already have `batch` loaded, it'd be easy to check if they are the same or not.

Not a big deal ofc.. but in case you want to fix this little inefficiency. 

You also said:

> (There is no migration. We can just edit the batches and submit it to the database, then the data should be fetched.)

This won't work if we make this change tho. We'd need to create a migration script.

Up to you!

phipsae replied:
Sorry, I overlooked it, will do it tomorrow.

phipsae commented:
Added migration script and changed the logic, that only NFT address is fetched, when batch registry contract addresses changes. (Somehow I messed up with the commits names).

carletex reviewed (approved):
Thanks for this Philip! 

Pushed https://github.com/scaffold-eth/buidlguidl-v3/pull/119/commits/63e72f7795d441f5780ec4db95efbba72c99dc7a so we can read the NFT address from the API endpoint. Also pushed a couple of little things.
